### PR TITLE
Add tests to openmrsUtils Trunk-5087

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/projects/TRUNK/issues/TRUNK-5087?filter=allopenissues
Added tests to openmrsUtils trunk-5087

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by sherrif10** [comment]
Again the travis didnot pass successfully but this case the travis didnot completely build. Am still wondering what could be the issue. @mozzy11 @dkayiwa

**Comment by HerbertYiga** [comment]
was this passing locally,according to your logs changes made at  OpenmrsUtilTest are the ones breaking the build.

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 1]
```diff
@@ -422,6 +424,33 @@ public void containsDigit_shouldReturnFalseIfStringContainsNoDigits() {
...
+	}
+
+	/**
+	 * @see Openmrs Checks whether the current JVM version matches with
+	 *      jdkversion specified
+	 */
+	@Test
+	public void JVMVersion_shouldReturnTrueIfOpenmrsVersionMatchesWithJdkVersion() {
+
+		if (JdkVersion.getJavaVersion().matches("1.(0|1|2|3|4|5).(.*)")) {
```
Comment (mozzy11) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: @sherrif10   what are you trying to do here?

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 2]
```diff
@@ -422,6 +424,33 @@ public void containsDigit_shouldReturnFalseIfStringContainsNoDigits() {
...
+
+	/**
+	 * @see OpenmrsUtil#containsonlyuppercase(String)
+	 */
+	@Test
+	public void containsUpperCase_shouldReturnIfStringContainsOnlyUpperCase() {
+
+		Assert.assertFalse(
+				OpenmrsUtil.containsUpperCaseOnly("this is the case"));
+
```
Comment (mozzy11) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: remove the unnecesary space here
as advised here  https://wiki.openmrs.org/display/docs/Java+Conventions

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 3]
```diff
@@ -71,9 +72,10 @@
 	public void runBeforeEachTest() throws Exception {
 		initializeInMemoryDatabase();
 		authenticate();
-		
+
 		Context.getAdministrationService().saveGlobalProperty(luhnGP);
 	}
+
```
Comment (mozzy11) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: why that space ?

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 4]
```diff
@@ -71,9 +72,10 @@
 	public void runBeforeEachTest() throws Exception {
 		initializeInMemoryDatabase();
 		authenticate();
-		
+
```
Comment (mozzy11) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: unnecesary space

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 5]
```diff
@@ -422,6 +424,33 @@ public void containsDigit_shouldReturnFalseIfStringContainsNoDigits() {
...
+	 */
+	@Test
+	public void JVMVersion_shouldReturnTrueIfOpenmrsVersionMatchesWithJdkVersion() {
+
+		if (JdkVersion.getJavaVersion().matches("1.(0|1|2|3|4|5).(.*)")) {
+			throw new APIException(
+					"OpenMRS requires Java 6, but is running under "
+							+ JdkVersion.getJavaVersion());
+		}
+
```
Comment (mozzy11) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: unnecesary space

**Comment by sherrif10** [comment]
Thanks @mozzy i was testing that method. but that method seems to be making
the travis failing

**Comment by sherrif10** [comment]
i created another PR everything was ok but i got the travis complaining
that it was not complete . what could lead the travis ci not to completely
build

**Comment by sherrif10** [comment]
Thanks @mozzy am trying to study how could travis ci not completely build. i think there is a difference between  not compeletely build and failing

**Comment by mozzy11** [comment]
@sharif10 also look at https://wiki.openmrs.org/display/docs/Pull+Request+Tips
and https://wiki.openmrs.org/display/docs/Using+Git

**Comment by mozzy11** [comment]
@sherrif10  , you have a compiler error at this line OpenmrsUtilTest.java:[435,15]  , check propely what you did

**Comment by HerbertYiga** [comment]
@sharif am seing you have two pull requests for one ticket kindly close one

**Comment by sherrif10** [comment]
I actually closed two tickets and i created another ticket on different
branch @herbertYiga

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by dkayiwa** [comment]
@sherrif10 did you look at the travis build failure? Or did you abandon this?

**Comment by sherrif10** [comment]
Am not currently working on this ticket, it was even closed

**Comment by mozzy11** [comment]
@sherrif10  can you then go ahead and close this PR  ???

**Pull request closed without merging**